src/plugins/intel_cpu/src/nodes/mvn.cpp

```diff
@@ -333,97 +244,617 @@ struct jit_uni_mvn_mean_variance_kernel_f32 : public jit_uni_mvn_mean_variance_k
-        size_t unroll_number = div_up(vec_num, unroll_size);
+        // r12, rax, rdx, rbp, r15, rcx and rdi is available
+        // r13 is available as no fill need for this layout
+        // reg_rt_shape is C
+        Xbyak::Reg64 reg_unroll_size = r12;
+        Xbyak::Reg64 reg_unroll_num = rax;
+        Xbyak::Reg64 reg_vector_num = rbp;
+        Xbyak::Reg64 reg_tail_num = r13;
+        // size_t unroll_size = 4;
+        mov(reg_unroll_size, 4);
+        // size_t vec_num = C / vector_step
+        mov(rax, reg_rt_shape);
+        mov(reg_vector_num, vector_step);
+        xor_(rdx, rdx);
+        div(reg_vector_num);    // reg_rt_shape / vector_step, rax is result, rdx is tails(yushu)
+        mov(reg_vector_num, rax);
+        mov(reg_tail_num, rdx);
+
+        Xbyak::Reg64 reg_src_aux = rdx;
+        Xbyak::Reg64 reg_work_amount_bk = r15;
+        mov(reg_work_amount_bk, reg_work_amount);  // should before tail jmp
+
+        Xbyak::Label tail_label;
+        cmp(reg_vector_num, 0);
+        je(tail_label, T_NEAR);
+
+        // unroll_size = vec_num >= unroll_size ? unroll_size : vec_num;
+        Xbyak::Label label_reset_unroll_size_end;
+        cmp(reg_unroll_size, reg_vector_num);
+        jle(label_reset_unroll_size_end, T_NEAR);
+        mov(reg_unroll_size, reg_vector_num);
+        L(label_reset_unroll_size_end);
+
+        // last unroll_size
+        Xbyak::Label label_reset_last_unroll_size;
+        Xbyak::Label label_reset_last_unroll_size_end;
+        Xbyak::Reg64 last_unroll_size = rcx;
+        mov(rax, reg_vector_num);
+        xor_(rdx, rdx);
+        div(reg_unroll_size);  // rdx
+        cmp(rdx, 0);
+        je(label_reset_last_unroll_size, T_NEAR);
+        mov(last_unroll_size, rdx);
+        jmp(label_reset_last_unroll_size_end);
+        L(label_reset_last_unroll_size);
+        {
+            mov(last_unroll_size, reg_unroll_size);
+        }
+        L(label_reset_last_unroll_size_end);
+
+        // size_t unroll_number = div_up(vec_num, unroll_size); --> (vec_num + unroll_size - 1) / unroll_size;
+        mov(rdi, reg_vector_num);
+        add(rdi, reg_unroll_size);
+        sub(rdi, 1);
+        mov(rax, rdi);
+        xor_(rdx, rdx);
+        div(reg_unroll_size);  // result is in rax, that is reg_unroll_num, no mov need.
 
         int ur_base = 4;
-        Xbyak::Reg64 reg_src_aux = reg_stride;
-        Xbyak::Reg64 reg_work_amount_bk = rbx;
-        mov(reg_work_amount_bk, reg_work_amount);
-        for (size_t ur_num = 0; ur_num < unroll_number; ur_num++) {
+        Xbyak::Label label_unroll_num;
+        Xbyak::Label label_unroll_num_end;
+        L(label_unroll_num);
+        {
+            cmp(reg_unroll_num, 0);
+            jle(label_unroll_num_end, T_NEAR);
+
+            Xbyak::Label label_not_last;
+            cmp(reg_unroll_num, 1);
+            jne(label_not_last, T_NEAR);
+            mov(reg_unroll_size, last_unroll_size);
```

Comment:
As I understatnd, code block here https://github.com/chenhu-wang/openvino/blob/chenhu/shape_agnostic_jit_kernel_for_MVN/src/plugins/intel_cpu/src/nodes/mvn.cpp#L279-L294 shows` last_unroll_size = reg_vector_num / reg_unroll_size`. And here we assign last_unroll_size back to reg_unroll_size again. Could you please help me understand the meaning of last_unroll_size for different cases?

Reply:
assume total shape is 16 *4 + 16 *4 + 16*2 +1. unroll with 4 vector. 16*2 is the last unroll, 1 is the tail